app/models/comment.rb

```diff
@@ -0,0 +1,17 @@
+class Comment < ApplicationRecord
+  belongs_to :user, optional: true
+  belongs_to :commentable, polymorphic: true
+  belongs_to :parent, optional: true, class_name: "Comment"
+
+  def comments
+    Comment.where(commentable: commentable, parent_id: id)
+  end
```

Comment:
Use has_many instead of instance method

Reply:
I  use this method for to get parent post and nesting reply comments